client/src/components/TreeDropdownField/tests/TreeDropdownField-test.js

```diff
@@ -1,684 +1,696 @@
   });
+  props = {
+    ...props,
+    actions: {
+      ...props.actions,
+      treeDropdownField: {
+        ...props.actions.treeDropdownField,
+        setVisible
+      }
+    },
+    value: 67,
+    data: {
+      valueObject: {
+        id: 67
+      },
+      urlTree: 'foo'
+    }
+  };
+  render(<TreeDropdownField {...props}/>);
+  await promise;
+  expect(setVisible).toBeCalledWith('Form_Test', [4]);
+});
 
-  describe('getVisibleTree()', () => {
-    it('should call findTreeByPath()', () => {
-      props.tree = {};
-      props.visible = [1];
-      props.findTreeByPath = jest.fn();
-
-      field = ReactTestUtils.renderIntoDocument(
-        <TreeDropdownField {...props} />
-      );
-
-      field.getVisibleTree();
-
-      expect(props.findTreeByPath).toBeCalledWith(props.tree, props.visible);
-    });
+test('TreeDropdownField single-select should add valueObject to selectedValues', () => {
+  const addSelectedValues = jest.fn();
+  let props = makeProps({
+    findTreePath: jest.fn((data, value) => (value ? [4, value] : null))
   });
+  props = {
+    ...props,
+    actions: {
+      ...props.actions,
+      treeDropdownField: {
+        ...props.actions.treeDropdownField,
+        addSelectedValues
+      }
+    },
+    value: 67,
+    data: {
+      valueObject: {
+        id: 67
+      },
+      urlTree: 'foo'
+    }
+  };
+  render(<TreeDropdownField {...props}/>);
+  expect(addSelectedValues).toBeCalledWith(props.id, [{id: 67}]);
+});
 
-  describe('getBreadcrumbs()', () => {
-    it('should traverse the path given and return the relevant nodes in the tree', () => {
-      props.visible = [5, 9, 26];
-      field = ReactTestUtils.renderIntoDocument(
-        <TreeDropdownField {...props} />
-      );
-      const breadcrumbs = field.getBreadcrumbs();
-
-      expect(breadcrumbs[0].id).toBe(props.visible[0]);
-      expect(breadcrumbs[1].id).toBe(props.visible[1]);
-      expect(breadcrumbs[2].id).toBe(props.visible[2]);
-    });
-
-    it('should traverse and terminate early if using a missing node', () => {
-      props.visible = [5, 8, 26];
-      field = ReactTestUtils.renderIntoDocument(
-        <TreeDropdownField {...props} />
-      );
-      const breadcrumbs = field.getBreadcrumbs();
-
-      expect(breadcrumbs[0].id).toBe(props.visible[0]);
-      expect(breadcrumbs[1]).toBeFalsy();
-    });
+test('TreeDropdownField single-select should not call selectedValues without a valueObject', () => {
+  const addSelectedValues = jest.fn();
+  let props = makeProps({
+    findTreePath: jest.fn((data, value) => (value ? [4, value] : null))
   });
+  props = {
+    ...props,
+    actions: {
+      ...props.actions,
+      treeDropdownField: {
+        ...props.actions.treeDropdownField,
+        addSelectedValues
+      }
+    },
+  };
+  render(<TreeDropdownField {...props}/>);
+  expect(addSelectedValues).not.toBeCalled();
+});
 
-  describe('lazyLoad()', () => {
```

Comment:
Even after spending several hours on this, I was simply unable to get lazyLoad() to trigger